modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTController.scala

```diff
@@ -22,10 +22,71 @@ import squants.time.Time
 final case class GHOSTController[F[_]: Sync](ghostClient: GHOSTClient[F],
                                              gdsClient: GDSClient) {
   import GHOSTController._
-  val log: Logger = getLogger
+  val Log: Logger = getLogger
+
+  // TODO: Are GIAPI parameters maintained between applies? Here is a use case that requires more information:
+  // 1. We observe in standard resolution mode using both IFUs on targets.
+  // 2. We observe in high resolution mode using only IFU1 on a target.
+  // Question: Do we need to, for (2), touch IFU2 at all (i.e. to standard mode, no target)?
+  // Do we need to always set target type for IFUs? How about bundle configuration?
+  //
+  // Right now, this only generate a config if the name, RA, and dec are defined: otherwise, we get the empty config.
+  private def ifuConfig(ifuNum: IFUNum,
+                        nameOpt: Option[String],
+                        raOpt: Option[HourAngle],
+                        decOpt: Option[Angle],
+                        bundleConfig: BundleConfig): Configuration = {
+    def cfg[P: Show](paramName: String, paramVal: P) =
+      Configuration.single(s"${ifuNum.ifuStr}.$paramName", paramVal)
+
+    val config = for {
+      _   <- nameOpt // We only do this because we want Configuration.Zero if there is no target name.
+      ra  <- raOpt
+      dec <- decOpt
+    } yield {
+      cfg("target", IFUTargetType.determineType(nameOpt).targetType) |+|
+        cfg("ra", ra.toDoubleDegrees) |+|
+        cfg("dec", dec.toDoubleDegrees) |+|
+        cfg("bundle", bundleConfig.configName)
+    }
+
+    config.getOrElse(Configuration.Zero)
+  }
+
+  private def srifuConfig(config: GHOSTConfig): Configuration =
+    ifuConfig(IFUNum.IFU1, config.srifu1Name, config.srifu1CoordsRAHMS,
+      config.srifu1CoordsDecDMS, BundleConfig.Standard) |+|
+      ifuConfig(IFUNum.IFU2, config.srifu2Name, config.srifu2CoordsRAHMS,
+        config.srifu2CoordsDecDMS, BundleConfig.Standard)
+
+  private def hrifuConfig(config: GHOSTConfig): Configuration =
+    ifuConfig(IFUNum.IFU2, config.hrifu1Name, config.hrifu1CoordsRAHMS,
+      config.hrifu1CoordsDecDMS, BundleConfig.HighRes) |+|
+      ifuConfig(IFUNum.IFU2, config.srifu2Name, config.srifu2CoordsRAHMS, config.srifu2CoordsDecDMS,
+        BundleConfig.HighRes)
+
+  // If the srifu parameters are defined, use them; otherwise, use the hrifu parameters.
+  // TODO: What do we do with the base position explicit override?
+  // TODO: This was not on the list of provided parameter names.
+  private def ghostConfig(config: GHOSTConfig): SeqActionF[F, CommandResult] = {
+    val giapiApply = srifuConfig(config) |+| hrifuConfig(config)
+
+    EitherT(ghostClient.genericApply(giapiApply).attempt)
```

Comment:
This feel like common code with `GPI` could we factor it out?

Reply:
Agreed. Maybe we need a `GIAPIInstrument` or something?

Reply:
Yeah, though we can do that in another PR